scripts/common.make

```diff
@@ -31,6 +31,27 @@ ifdef WINDOWS
 WINDOWS_BUILD=1
 endif
 
+## Usage: $(call remove_unset_options,PREPROCESSOR_INPUT)
+## Remove the preprocessor symbols that are not set in the current configuration
+## from PREPROCESSOR_INPUT. Also normalize whitespace.
+## Example:
+##   $(call remove_unset_options,MBEDTLS_FOO MBEDTLS_BAR)
+## This expands to an empty string "" if MBEDTLS_FOO and MBEDTLS_BAR are both
+## disabled, to "MBEDTLS_FOO" if MBEDTLS_BAR is enabled but MBEDTLS_FOO is
+## disabled, etc.
+##
+## This only works with a Unix-like shell environment (Bourne/POSIX-style shell
+## and standard commands) and a Unix-like compiler (supporting -E). In
+## other environments, the output is likely to be empty.
+define remove_unset_options
+$(strip $(shell
```

Comment:
I am suspecting that we are spinning up a shell here, but would like to be sure. Could you please give me a link to a reference manual that explains what the `shell` command does.

Reply:
[`shell`](https://www.gnu.org/software/make/manual/html_node/Shell-Function.html#index-shell) and [`strip`](https://www.gnu.org/software/make/manual/html_node/Text-Functions.html#index-strip) are functions in the GNU make language.